### ISSUE TITLE
Matmul and Size error on Windows

### ISSUE THREAD

**Issue by peterjc123** [pytorch/pytorch#4983]
- OS: Windows
- PyTorch version: master
- How you installed PyTorch (conda, pip, source): source
- Python version: 3.6
- CUDA/cuDNN version: 9.0
- GPU models and configuration: 1070
- GCC version (if compiling from source): MSVC 15.1

It seems that #4943 breaks Windows build with the following error:
```pytb
======================================================================
ERROR: test_broadcast_batched_matmul (__main__.TestTorch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_torch.py", line 1625, in test_broadcast_batched_matmul
    self._test_broadcast_batched_matmul(self, lambda t: t)
  File "test_torch.py", line 1622, in _test_broadcast_batched_matmul
    verify_batched_matmul(*indices)
  File "test_torch.py", line 1606, in verify_batched_matmul
    truth = maybe_squeeze_result(lhs_expanded, rhs_expanded, lhs_expanded_matmul_fn(rhs_expanded))
  File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test\test\torch\tensor.py", line 196, in matmul
    return torch.matmul(self, other)
  File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test\test\torch\functional.py", line 264, in matmul
    expand_batch_portion = torch._C._infer_size(tensor1_exp_size[:-2], tensor2_exp_size[:-2])
RuntimeError: expected a torch.Size as argument 1

======================================================================
ERROR: test_matmul_out (__main__.TestTorch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_torch.py", line 1642, in test_matmul_out
    check_matmul((2, 3, 4), (2, 4, 5))
  File "test_torch.py", line 1632, in check_matmul
    expected = torch.matmul(a, b)
  File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test\test\torch\functional.py", line 264, in matmul
    expand_batch_portion = torch._C._infer_size(tensor1_exp_size[:-2], tensor2_exp_size[:-2])
RuntimeError: expected a torch.Size as argument 1

======================================================================
FAIL: test_Size (__main__.TestTorch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_torch.py", line 5293, in test_Size
    self.assertIsInstance(x * 2, torch.Size)
AssertionError: (1, 2, 3, 1, 2, 3) is not an instance of <class 'torch.Size'>
```

It seems that multipling `torch.Size` with sth make it a tuple, not `torch.Size`.

**Comment by yf225** [comment]
Yes it's a known issue, @colesbury is looking at it with https://github.com/pytorch/pytorch/pull/4980.

**Comment by peterjc123** [comment]
Resolved by #4980 and #4991.
